community-tutorials/high-availability-keepalived/01-en.md

```diff
@@ -0,0 +1,244 @@
+- *YOUR\_WEBSERVICE\_PASSWORD* is the one we just set in step 2.1.4
+- *YOUR\_FAILOVER\_IP* can be found in your CCP in the section *Products* or in the SCP in the *Network* section
+- *YOUR\_VSERVER\_NAME* can be found in your SCP in the *General* section of the corresponding server
+- *YOUR\_VSERVER\_MAC* can also be found in your SCP in the *General* section of the corresponding server (Note that you need the MAC of your primary interface not the MAC of the vLAN)
+
+# Step 3 - Setup keepalived
+## Step 3.1 - Install keepalived
+Navigate to either the website [https://keepalived.org/download.html](https://keepalived.org/download.html) or the respective GitHub repository [https://github.com/acassen/keepalived](https://github.com/acassen/keepalived) and install the application as per the instructions available at [https://github.com/acassen/keepalived/blob/master/INSTALL](https://github.com/acassen/keepalived/blob/master/INSTALL).
+## Step 3.2 - Turn on Packet forwarding
+In order for the keepalived service to forward packets you need to turn on forwarding in the kernel. Edit `/etc/sysctl.conf` and make sure to include the following line using your favourite text editor e.g. `sudo vim /etc/sysctl.conf`:
+```
+net.ipv4.ip_forward = 1
+```
+For the changes to take effect execute `sudo sysctl -p`
+
+## Step 3.3 - Configure keepalived
+I will provide a basic configuration here which is a good starting point for further tweaks if needed. With this configuration s1 starts as master and s2 starts as backup. If s1 is for any reason not sending adverts anymore / s2 does not receive adverts anymore, s2 will automatically jump in as a failover.
+
+One interesting parameter to start tweaking is the *advert\_int* parameter. This parameter on the one hand determines how often an advert should be sent (every 2 seconds in this case) and on the other hand determines that if after 3 * *advert\_int* seconds no advert was received the failover is triggered (so in this case; after 3 * 2 = 6 seconds).
+
+If a server changes from BACKUP to MASTER state the *notify\_master* script, which is our *to\_master.sh* (that reroutes our Failover IPv4), is triggered.
+
+But there are much more methods you can use to trigger a failover besides the adverts. You could for example also track files, scripts, processes, interfaces etc. to increase/decrease the priority and trigger the failover based on this. Especially scripts arev very flexible and offer the possibility for more sophisticated checks. If you are interested in more configuration methods this [Red Hat Article](https://www.redhat.com/sysadmin/advanced-keepalived) covers some configurations.
+
+**Example */etc/keepalived/keepalived.conf* for s1**
+```
+vrrp_instance VI_1 {
+    state MASTER
+    interface eth1
+    virtual_router_id 101
+    priority 255
+    advert_int 2
+    authentication {
+        auth_type PASS
+        auth_pass CUSTOM_PASSWORD
+    }
+    virtual_ipaddress {
+        10.132.0.100
```

Comment:
Which IP address are we referring here?

Reply:
That is the virtual ip address of our vrrp master. So at any time either s1 or s2 owns this address. We currently do not use it but if we had other servers in our vlan we could also use this IP to connect to the master server.